docs/source/developers/release.rst

```diff
@@ -87,12 +87,55 @@ These are the different steps that are required to create a release candidate.
     # Curate the release
     # The end of the generated report shows the jira tickets with wrong version number assigned.
     archery release curate <version>
-    
-    # Checkout release branch
-    # Use master for major releases
-    git checkout -b release-4.0.0 master
-    # Use maintenance branches like maint-4.0.x for patch releases
-    git checkout -b release-4.0.1 maint-4.0.x
+
+
+Creating a Release Candidate
+============================
+
+These are the different steps that are required to create a release candidate.
+
+For the initial Release Candidate, we will create a maintenance branch from master.
+Follow up Release Candidates will update the maintenance branch cherry-picking
+specific commits.
+
+We have implemented a Feature Freeze policy between Release Candidates.
+This means that, in general, we will only add fixes between Release Candidates.
+In rare cases, critical features can be added between release candidates, if
+there is community consensus.
+
+Create or update the corresponding maintenance branches:
+
+.. tab-set::
+
+   .. tab-item:: Initial Release Candidate
+
+      .. code-block::
+
+            # Execute the following from an up to date master branch.
+            # This will create a branch locally called maint-X.Y.Z
```

Comment:
What is "X.Y.Z" supposed to stand for? It's not clear to me.

Reply:
I've added a minor comment, let me know what you think:
```
            # Execute the following from an up to date master branch.
            # This will create a branch locally called maint-X.Y.Z
            # X.Y.Z corresponds with the Major, Minor and Patch version number
            # of the release respectively. As an example 9.0.0
```

Reply:
I like `maint-9` for all 9.Y.Z releases.
I assume that we don't need multiple maintenance branches for the same major version. In other words, we don't release 9.0.1 after we release 9.1.0.

Reply:
I am not sure about that. We have had discussions in the mailing list about SEMVER and trying to have more releases. I do think being prepared with a versioning naming of Major.Minor.Patch on our maintenance branches that is flexible enough is not a problem and might benefit us in the future in order to achieve these other goals.